code/game/machinery/doors/firedoor.dm

```diff
@@ -23,12 +23,89 @@
 	interaction_flags_machine = INTERACT_MACHINE_WIRES_IF_OPEN | INTERACT_MACHINE_ALLOW_SILICON | INTERACT_MACHINE_OPEN_SILICON | INTERACT_MACHINE_REQUIRES_SILICON | INTERACT_MACHINE_OPEN
 	var/nextstate = null
 	var/boltslocked = TRUE
-	var/list/affecting_areas
 	var/being_held_open = FALSE
+	/// The merger id used to create/get the merger group in charge of handling tanks that share an internal gas storage
+	var/merger_id = "firelock"
+	/// The typecache of types which are allowed to merge internal storage
+	var/static/list/merger_typecache
```

Comment:
And i'm pretty sure `typecacheof` has it's own internal cache. Storing it here would be redundant.

Reply:
For this I just followed what the tanks do as ninja made them, I'm not sure if removing this will work but I can try

Reply:
Disregard above. Still, it's called infrequently enough that I don't think it warrants caching.

Reply:
i removed the static list, and just typecache directly in the proc

Reply:
> 
> 
> Disregard above. Still, it's called infrequently enough that I don't think it warrants caching.

It's used in what amounts to reaction to atmos process, which while it's not hyper hot seems like a bit of a waste. We're also making one list per group of firelocks right now